src/middleware.ts

```diff
@@ -0,0 +1,32 @@
+import { NextRequest, NextResponse } from "next/server";
+import { client } from "../studio/lib/client";
+import { RedirectSparsePage } from "../studio/lib/payloads/redirect";
+
+export async function middleware(request: NextRequest) {
+  const slug = request.nextUrl.pathname;
+  const redirect = await client.fetch<RedirectSparsePage | null>(
+    `*[_type == "redirect" && source.current == "${slug}"][0]{
+        "destination":destination.current,
+        permanent
+      }`,
+  );
+  if (redirect !== null) {
+    return NextResponse.redirect(
+      new URL(redirect.destination, request.url),
+      redirect.permanent ? 308 : 307,
```

Comment:
```suggestion
      redirect.permanent ? permanentRedirect : temporaryRedirect,
```
Just because not everyone remembers http status codes at the top of their head 😅 so add them as consts outside the middleware function?